chore: add prettier format command

netbonus opened the pull request.

netbonus committed:
chore: add prettier

netbonus committed:
fix: exclude JSONC test vectors from prettier formatting

The vectors.jsonc file contains test data with specific formatting
that should be preserved. Added .prettierignore to prevent automatic
formatting of this file.

yilmazbahadir commented on package.json:
```diff
@@ -8,6 +8,7 @@
     "commit": "git-cz",
     "lint:fix": "eslint src --c .ts,.tsx --config eslint.config.mjs --fix",
     "lint": "eslint src --c .ts,.tsx --config eslint.config.mjs",
+    "format": "prettier --write .",
```
Should we use eslint-config-prettier to disable ESLint rules that conflict with Prettier, and let lint:fix handle both?

```
  pnpm add -D eslint-config-prettier
```
And that way, precommit can auto-fix the format issues.

yilmazbahadir replied:
I see now `eslint-config-prettier` is already configured. So `lint: fix` should already format the ts and tsx files in the src. Maybe we should do the following change to include others as well:
` "lint:fix": "eslint src --c .ts,.tsx --config eslint.config.mjs --fix && prettier --write ." `

netbonus replied:
added

yilmazbahadir replied:
cannot see -- maybe the change got stuck?

netbonus committed:
refactor: enable eslint for test directories and fix type safety issues

Removes eslint ignores for e2e, integration, and utils test directories to improve code quality.
Fixes all resulting linting errors including duplicate imports, missing type definitions, and empty interfaces.

netbonus committed:
chore: integrate prettier into lint:fix workflow

Updates lint:fix to run both eslint and prettier, addressing PR feedback.
This ensures all files (not just TypeScript) are formatted during linting.

Also excludes pnpm-lock.yaml from prettier formatting as it's a generated file.
